### DISCUSSION TITLE
GPU implementation of barrier functions

### DISCUSSION THREAD

**Discussion by antoinebou12** [ipc-sim/ipc-toolkit#170] [Ideas]
**Is your feature request related to a problem? Please describe.**

GPU implementation of barrier functions

**Additional context**
https://kemenghuang.github.io/img/ugipc-arxiv.pdf

**Comment by MuGdxy** [comment]
The GPU version can be found here [libuipc](https://github.com/spiriMirror/libuipc), in the source code of [backend cuda](https://github.com/spiriMirror/libuipc/blob/main/src/backends/cuda/contact_system/contact_models/codim_ipc_simplex_normal_contact_function.h).
